cabal-install/Distribution/Client/ProjectConfig.hs

```diff
@@ -401,9 +404,16 @@ readProjectLocalConfig verbosity projectRootDir = do
         projectPackages         = [ "./*.cabal" ],
 
         -- This is to automatically pick up deps that we unpack locally.
-        projectPackagesOptional = [ "./*/*.cabal" ]
+        projectPackagesOptional = [ "./*/*.cabal" ],
+
+        projectConfigProvenance = Set.singleton Implicit
       }
 
+    setExplicitProjectConfig :: ProjectConfig -> ProjectConfig
+    setExplicitProjectConfig config =
```

Comment:
When you say 'helper function' I'm going to assume you mean moving it to the top-level, which since it's not being used elsewhere it's nice (IMO) to have that apparent by its location in a where clause.

The old provenance should indeed be preserved via `Set.insert`.